File: day2/puzzle2.py
import re

total = 0
for l in open('day2.txt'):
 m = re.match('([0-9]+)x([0-9]+)x([0-9]+)', l)
 l = int(m.group(1))
 w = int(m.group(2))
 h = int(m.group(3))
 lw = 2*(l+w)
 wh = 2*(w+h)
 hl = 2*(h+l)
 wrap = min(lw,wh,hl)
 ribbon = l * w * h
 total = total + wrap + ribbon
 print l, w, h, "=", wrap, ribbon 
print "total", total

File: day2/puzzle2.js
var fs = require('fs');

fs.readFile('day2.txt', 'utf-8', function (err, inp) {
 if (err) throw err;
 console.log(inp.trim().split(/\n/g).reduce(function (curr, line) {
  var m = line.match(/([0-9]+)x([0-9]+)x([0-9]+)/);
  var l = Number(m[1]), w = Number(m[2]), h = Number(m[3]);
  var wrap = Math.min(2*(l+w),2*(w+h),2*(h+l)), ribbon = l*w*h
  return curr + wrap + ribbon;
 }, 0));
});
